### PR TITLE
Switch to C++17

### PR DESCRIPTION
**Pull request by teams2ua**

### CODE REVIEW

File: core/src/CMakeLists.txt [hunk 1]
```diff
@@ -49,6 +49,7 @@ if(IS_IOS GREATER_EQUAL 0)
             ${HEADERS_FILES})
     target_link_libraries(ledger-core PUBLIC ledger-core-interface)
     set(CMAKE_SHARED_LINKER_FLAGS "-Wall")
+    set(CMAKE_CXX_FLAGS "-fno-aligned-allocation")
```
Comment (phaazon) [inline core/src/CMakeLists.txt]: This is not related to C++17. Why are you doing this?

Comment (teams2ua) [reply core/src/CMakeLists.txt]: C++17 adds aligned allocations, but ARM doesn't support it, so to make the iOS build works I need to do this, in case some library will use this feature (I am looking at you stlab)

Comment (phaazon) [reply core/src/CMakeLists.txt]: Roger that. Can you then make that setting depends on ARM architecture, so that he doesn’t leak to other architectures as well?

**Comment by teams2ua** [comment]
Postponed until we start include libc.so to linux distributions of ledger-live
